### ISSUE TITLE
E1101 false positives for exception variables

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#32]
Originally reported by: **Chris Rebert (BitBucket: [cvrebert](http://bitbucket.org/cvrebert), GitHub: @cvrebert?)**

----------------------------------------

Testcase:
```
#!python
# example.py
x = []

try:
    x[42]
except IndexError as err:
    err.message.find("out of range")
```
Somehow "message" ends up being treated as a class and the inaccurate warning gets repeated 5 times for some reason:
```
#!bash
$ pylint --include-ids=y --report=n example.py 
No config file found, using default configuration
************* Module example
C0111:  1,0: Missing docstring
C0103:  1,0: Invalid name "x" for type constant (should match (([A-Z_][A-Z0-9_]*)|(__.*__))$)
E1101:  6,4: Class 'message' has no 'find' member
E1101:  6,4: Class 'message' has no 'find' member
E1101:  6,4: Class 'message' has no 'find' member
E1101:  6,4: Class 'message' has no 'find' member
E1101:  6,4: Class 'message' has no 'find' member
$ pylint --version
No config file found, using default configuration
pylint 0.28.0, 
astng 0.24.3, common 0.59.1
Python 2.7.5 (default, May 21 2013, 11:38:15) 
[GCC 4.2.1 Compatible Apple Clang 4.0 ((tags/Apple/clang-421.0.60))]
```

----------------------------------------
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/32

**Comment by PCManticore** [comment]
By the way, this is solved in astroid with https://github.com/PyCQA/astroid/commit/a315ef29ff51ff861cb5111d706bc17d6375f0de, but it's going to work only for Pylint 2.0, so I'm leaving this issue open for reference.

**Comment by PCManticore** [comment]
Needs backport of the special objects from astroid, which I'm planning to do for 1.6.0/astroid 1.5 anyway. We don't need to wait for 2.0 for it.

**Comment by PCManticore** [comment]
This is now fixed, since I backported in astroid's master branch the required special model changes. Will be available with the next release.
